dashboard/views.py

```diff
@@ -150,3 +152,36 @@ def post(self, request):
         return Response(
             data=enrollment.json
         )
+
+class UserPaymentStatus(APIView):
+    """
+    returns list of program id and status weather user has paid for atleast one course in selected program.
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+        authentication.TokenAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def get(self, request):
+        data = []
+        for program_enrollment in ProgramEnrollment.objects.filter(
+                user=request.user
+        ).iterator():
```

Comment:
We should use `select_related` here to avoid extra queries to `User` and `Program`. 

```python
for program_enrollment in ProgramEnrollment.objects.filter(user=request.user).select_related('user', 'program').iterator()
```